lib/internal/errors.js

```diff
@@ -413,6 +407,16 @@ function uvErrmapGet(name) {
   return uvBinding.errmap.get(name);
 }
 
+const captureLargerStackTrace = hideStackFrames(
+  function captureLargerStackTrace(err) {
+    userStackTraceLimit = Error.stackTraceLimit;
+    Error.stackTraceLimit = Infinity;
```

Comment:
During module resolution, the TypeScript compiler has to probe many paths that turn out not to exist.  To mitigate the cost of creating so many `uvExceptions`, it sets `Error.stackTraceLimit` to zero (helpfully suggested [here](https://github.com/nodejs/node/pull/33716#pullrequestreview-424029712)).  From profiling, it looks like this change (specifically `captureLargerStackTrace node:internal/errors:450`) resulted in a pretty substantial perf regression.  Fortunately, newer version of TypeScript use the [new](https://github.com/nodejs/node/pull/33716/files) `throwIfNoEntry` flag, mitigating the issue, but older versions of TypeScript are likely to be substantially affected.  Have others reported performance issues in this area and, if so, would it be worthwhile to file a bug (I couldn't find an existing one)?

Reply:
You should be able to hack your way around it in Node.js v16.0.0+ by making the `stackTraceLimit` not writable:
```js
Reflect.defineProperty(Error, 'stackTraceLimit', {
  writable: false,
  value: 0,
});
```

Reply:
I have to admit, that would not have occurred to me. :smile:

Fortunately, we already have a good mitigation for v16 - we just pass `throwIfNoEntry`.  The problem is more that code we've already shipped (i.e. TypeScript <= 4.1) lacks either mitigation and users will suddenly see a large slowdown.  In the repro I happened to notice this in (which, unfortunately, I can't share), the runtime jumps from 30 seconds to 48 seconds.

(If we end up filing a bug for this, I'll make sure to include a publicly-available sample.)

Reply:
Ideally, we would have detected it's set to zero and not increased it in that case and only would have set it to it's current limit + a small number for the extra frames that should be removed.